### PR TITLE
Tylors/ident c179 facebookoauth button

### PR DESCRIPTION
**Pull request by TylorS**
Manditory Checks
---

- [x] I added new tests for the issue I fixed or the feature I built
- [x] I ran `npm test` for the package I'm modifying
- [x] I used `npm run commit` instead of `git commit`

---

### CODE REVIEW

**Comment by TylorS** [comment]
[IDENT: Facebook oauth button](https://trello.com/c/uF5P17xs/171-ident-facebook-oauth-button)

File: src/components/FacebookAuthenticationButton/index.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  const view$: Stream<VNode> = just(view);
+
+  return {
+    dom: view$,
+    authentication$,
+  };
+}
+
+const getRedirectResult: AuthenticationType =
```
Comment (Frikki) [inline src/components/FacebookAuthenticationButton/index.ts]: `getRedirectResult` implies that it is an action.

File: src/components/FacebookAuthenticationButton/index.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    method: REDIRECT,
+    provider: new firebase.auth.FacebookAuthProvider(),
+  };
+
+export const facebookButtonId = `#facebook-auth`;
+
+const classNames: string =
+  '.c-btn--large';
+
+const style =
```
Comment (Frikki) [inline src/components/FacebookAuthenticationButton/index.ts:43]: What’s this?!?

File: src/components/FacebookAuthenticationButton/index.ts (hunk 1 above)
Comment (TylorS) [reply src/components/FacebookAuthenticationButton/index.ts]: It is an action of getting the redirect result from firebase, though.

File: src/components/FacebookAuthenticationButton/index.ts (hunk 2 above)
Comment (TylorS) [reply src/components/FacebookAuthenticationButton/index.ts:43]: It's hard to tell which part of the code your comment is referring to. Is it the `style` ?

File: src/components/FacebookAuthenticationButton/index.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+const facebookRedirectAuthentication: AuthenticationType =
+  {
+    method: REDIRECT,
+    provider: new firebase.auth.FacebookAuthProvider(),
+  };
+
+export const facebookButtonId = `#facebook-auth`;
+
+const classNames: string =
+  '.c-btn--large';
```
Comment (Frikki) [inline src/components/FacebookAuthenticationButton/index.ts:41]: Where did you find the information that the button should have a large modifier?

Comment (TylorS) [reply src/components/FacebookAuthenticationButton/index.ts:41]: The "make it look like the design" part of the cards

File: src/components/FacebookAuthenticationButton/index.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import { Stream, constant, just, startWith } from 'most';
+import { VNode, button } from '@motorcycle/dom';
+import { REDIRECT, GET_REDIRECT_RESULT, AuthenticationType }
+  from '../../drivers/firebase-authentication';
+import {
```
Comment (Frikki) [inline src/components/FacebookAuthenticationButton/index.ts]: Why are these imported? Shouldn’t they live in this file? (Sorry, Tylor. I didn’t make these rules. They’ll be killing us.)

File: src/components/FacebookAuthenticationButton/types.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { Stream } from 'most';
```
Comment (Frikki) [inline src/components/FacebookAuthenticationButton/types.ts]: This looks like it have been factored out. According to @sdebaun , we aren’t allowed to do such, unless I have misunderstood the Rules of Play. This is the whole file, I’m referring to.

**Review by Frikki — COMMENTED** [review]
A few issues needs feedback or corrections.

File: src/components/FacebookAuthenticationButton/index.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  FacebookAuthenticationButtonSources,
+  FacebookAuthenticationButtonSinks,
+} from './types';
```
Comment (Frikki) [inline src/components/FacebookAuthenticationButton/index.ts]: Why are these imported? Shouldn’t they live in this file? (Sorry, Tylor. I didn’t make these rules. They’ll be killing us.)

File: src/components/FacebookAuthenticationButton/index.ts (hunk 1 above)
Comment (Frikki) [reply src/components/FacebookAuthenticationButton/index.ts]: It looks like an object with a method. Not an action.

File: src/components/FacebookAuthenticationButton/index.ts (hunk 3 above)
Comment (Frikki) [reply src/components/FacebookAuthenticationButton/index.ts:41]: Sadly so.

File: src/components/FacebookAuthenticationButton/index.ts (hunk 2 above)
Comment (Frikki) [reply src/components/FacebookAuthenticationButton/index.ts:43]: Right. I need to add my comments below what I am referring to. I mean, where does this style come from?

**Review by Frikki — COMMENTED** [review]
Moved comment to better reflect its reference.

File: src/components/FacebookAuthenticationButton/index.ts (hunk 2 above)
Comment (TylorS) [reply src/components/FacebookAuthenticationButton/index.ts:43]: I used a color dropper extension on the image of the face book design to extract the colors.

File: src/components/FacebookAuthenticationButton/index.ts (hunk 1 above)
Comment (TylorS) [reply src/components/FacebookAuthenticationButton/index.ts]: Renamed.

File: src/components/FacebookAuthenticationButton/types.ts (hunk 5 above)
Comment (TylorS) [reply src/components/FacebookAuthenticationButton/types.ts]: Fixed.

File: src/components/FacebookAuthenticationButton/index.ts (hunk 6 above)
Comment (TylorS) [reply src/components/FacebookAuthenticationButton/index.ts]: Fixed.

**Comment by TylorS** [comment]
Made requested changes.

**Review by Frikki — APPROVED** [review]
I think this adheres to the Rules of Play.

**Pull request merged**